File: fpinjava-parent/fpinjava-lists-exercises/src/main/java/com/fpinjava/lists/exercise05_18/DoubleToString.java
package com.fpinjava.lists.exercise05_18;

import com.fpinjava.lists.exercise05_16.List;

public class DoubleToString {

  public static List<String> doubleToString(List<Double> list) {
    return List.foldRight(list, List.<String>list(), head -> tail -> tail.cons(Double.toString(head)));
  }
}

File: fpinjava-parent/fpinjava-lists-exercises/src/main/kotlin/com/fpinkotlin/lists/exercise05_18/DoubleToString.kt
package com.fpinkotlin.lists.exercise05_18

import com.fpinkotlin.lists.exercise05_16.List
import com.fpinkotlin.lists.exercise05_16.list

fun doubleToString(list: List<Double>): List<String> =
		list.foldRight(list<String>()) { head -> { tail -> tail.cons(head.toString()) } }
